elements/pl-order-blocks/dag_checker.py

```diff
@@ -45,27 +46,92 @@ def check_grouping(order: list[str], group_belonging: Mapping[str, Optional[int]
     return len(order)
 
 
-def grade_dag(order: list[str], depends_graph: Mapping[str, list[str]], group_belonging: Mapping[str, Optional[int]]) -> tuple[int, int]:
+def dag_to_nx(depends_graph: Mapping[str, list[str]]) -> nx.DiGraph:
+    """Convert input graph format from HTML into NetworkX object to utilize their algorithms."""
+    graph = nx.DiGraph()
+    for node in depends_graph:
+        for node2 in depends_graph[node]:
+            graph.add_edge(node2, node)
+    return graph
+
+
+def grade_dag(order: list[str], depends_graph: Mapping[str, list[str]], group_belonging: Mapping[str, Optional[int]]) -> int:
     """In order for a student submission to a DAG graded question to be deemed correct, the student
     submission must be a topological sort of the DAG and blocks which are in the same pl-block-group
     as one another must all appear contiguously.
     :param order: the block ordering given by the student
     :param depends_graph: The dependency graph between blocks specified in the question
     :param group_belonging: which pl-block-group each block belongs to, specified in the question
-    :return: length of list that meets both correctness conditions, starting from the beginning,
-    index of first wrong (-1 if none wrong)
+    :return: length of list that meets both correctness conditions, starting from the beginning
     """
-    graph = nx.DiGraph()
-    for node in depends_graph:
-        for node2 in depends_graph[node]:
-            graph.add_edge(node2, node)
+    graph = dag_to_nx(depends_graph)
 
     if not nx.is_directed_acyclic_graph(graph):
         raise Exception('Dependency between blocks does not form a Directed Acyclic Graph; Problem unsolvable.')
 
     top_sort_correctness = check_topological_sorting(order, graph)
     grouping_correctness = check_grouping(order, group_belonging)
 
-    correctness = top_sort_correctness if top_sort_correctness < grouping_correctness else grouping_correctness
-    first_wrong = -1 if correctness == len(order) else correctness
-    return correctness, first_wrong
+    return top_sort_correctness if top_sort_correctness < grouping_correctness else grouping_correctness
+
+
+def is_vertex_cover(G: nx.DiGraph, vertex_cover: Iterable):
+    """ this function from
+    https://docs.ocean.dwavesys.com/projects/dwave-networkx/en/latest/_modules/dwave_networkx/algorithms/cover.html#is_vertex_cover
+    """
```

Comment:
```suggestion
    """Determines whether the given set of vertices is a vertex cover of graph G.
    A vertex cover is a set of vertices such that each edge of the graph is incident with at least one vertex in the set.
    This function is taken from:
    https://docs.ocean.dwavesys.com/projects/dwave-networkx/en/latest/_modules/dwave_networkx/algorithms/cover.html#is_vertex_cover
    :param G: the graph on which to check the vertex cover
    :param vertex_cover: the iterable of nodes
    :return: True if the given iterable forms a vertex cover
    """
```
It might be good to mirror part of the docstring for clarity, in addition to the attribution link.

Reply:
I generally don't like having documentation that is longer than the function itself, because I think it just gets in the way of reading and understanding the code.

But if other people really want it I can put it in.